pkg/csi/cinder/openstack/openstack_volumes.go

```diff
@@ -219,6 +227,10 @@ func (os *OpenStack) WaitDiskDetached(instanceID string, volumeID string) error
 	err := wait.ExponentialBackoff(backoff, func() (bool, error) {
 		attached, err := os.diskIsAttached(instanceID, volumeID)
 		if err != nil {
+			if cpoerrors.IsNotFound(err) {
```

Comment:
ditto , could we handle in controller to adhere to csi spec?

Reply:
+1